src/components/Animation/CreateAnimation.vue

```diff
@@ -535,140 +752,177 @@ export default {
           ctx,
           timeLayer.title.slice(2),
           0.01 * infoCanvas.width + offsetX,
-          hPos,
+          hPos - 6,
           widths[0],
           30
         );
-        hPos += timeLayer.fontSize + 8;
+        hPos += baseFont + 8;
       });
 
       ctx.strokeStyle = "black";
       ctx.fillStyle = "black";
 
-      if (this.displayReferenceTime) {
-        canvasTxt.fontSize = 10;
-        canvasTxt.align = "left";
+      if (numModelRuns !== 0) {
+        canvasTxt.fontSize = baseFont - 2;
         ctx.font = canvasTxt.fontSize + "px sans-serif";
-        metrics = ctx.measureText(this.modelRunMessage[0]);
-        for (let i = 1; i < this.modelRunMessage.length; i++) {
-          let textLength = ctx.measureText(this.modelRunMessage[i]);
-          if (textLength.width > metrics.width) {
-            metrics = textLength;
+        canvasTxt.align = "left";
+        const allEqualOrEmpty = (arr) => {
+          const nonEmptyArr = arr.filter((v) => v !== "");
+          metrics = ctx.measureText(`Ref${this.$t("Colon")}` + nonEmptyArr[0]);
+          return nonEmptyArr.every((v) => v === nonEmptyArr[0]);
+        };
+        if (!allEqualOrEmpty(modelRuns)) {
+          for (let i = 1; i < modelRuns.length; i++) {
+            let textLength = ctx.measureText(
+              `Ref${this.$t("Colon")}` + modelRuns[i]
+            );
+            if (textLength.width > metrics.width) {
+              metrics = textLength;
+            }
           }
-        }
-
-        let MRPlacement = ctx_w - metrics.width - 0.01 * infoCanvas.width;
-        let side = 1;
-        if (!this.isLayerListShown) {
-          MRPlacement = 0.01 * infoCanvas.width;
-          side = 0;
-        }
-        for (let i = 0; i < this.modelRunMessage.length; i++) {
-          canvasTxt.drawText(
-            ctx,
-            this.modelRunMessage[i],
-            MRPlacement,
-            16 + 12 * i,
-            widths[side],
-            20
-          );
+          let MRPlacement = ctx_w - metrics.width - 0.01 * infoCanvas.width;
+          let side = 1;
+          if (!this.isLayerListShown) {
+            MRPlacement = 0.01 * infoCanvas.width;
+            side = 0;
+          }
+          let hPos = 0;
+          for (let i = 0; i < modelRuns.length; i++) {
```

Comment:
Likewise, change to use a `const` variable to avoid having to recalculate `.length` each loop.